language-design.md

```diff
@@ -28,6 +47,13 @@ you know a poorly designed language? What are the symptoms?
 
 **Response**
 
+Probably the most important and overarching symptom of a good programming language was discussed in class, and captured very well in a quote from the reading:
+> It should be easy to do simple things; possible to do complex things; and impossible, or at least difficult, to do wrong things. [Bloch 2006]
+
+All three of these are very important. If it is not easy to do simple things in a language, then users will never be willing to put in unnecessary work to do what would be easy in another language. The things that are categorized as simple may vary from language to language (or domain to domain), but there will always be simple operations that users understand should be easy. If complex things are impossible, then no one will ever be able to write more than a simple program; this may be acceptable for some little languages but is in general a sign that more features are necessary. Finally, most importantly for DSLs, the language needs to give the user a good sense of what things they should not do. If this is not the case then the language is likely poorly defined.
+
```

Comment:
Out of curiosity, how does this fit in with your response to the Pavlus article? How do you define what should be simple to do and what shouldn't? How simple does it need to be? I like the idea that the language should indicate what the language is capable of and where it's strengths are.